### ISSUE TITLE
need docs for optimize_append_index

### ISSUE THREAD

**Issue by DanRoscigno** [ClickHouse/ClickHouse#44799]

**Comment by DanRoscigno** [comment]
https://github.com/ClickHouse/ClickHouse/pull/18787

**Comment by rfraposa** [comment]
@CurtizJ @nikvas0 - I'm trying to figure out what `optimize_append_index` does. Can you summarize it for me? Here's the [pull request](https://github.com/ClickHouse/ClickHouse/commit/496d8ff46cfa6012964027532be3145cd30e2fea#diff-5773dfd8f761e06eb84c585f925923c8135240f59d3bdc38493c8bb0708b2298)

**Comment by rfraposa** [comment]
The PR mentioned above actually introduced three new settings. I added them in this PR: https://github.com/ClickHouse/ClickHouse/pull/45589
